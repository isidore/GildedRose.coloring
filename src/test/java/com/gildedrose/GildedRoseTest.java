package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.reporters.macosx.DiffMergeReporter;
import org.junit.Test;

@UseReporter(DiffMergeReporter.class)
public class GildedRoseTest {

	@Test
	public void testStuff() throws Exception {
		String[] names = { "name", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert",
				"Sulfuras, Hand of Ragnaros", "", null };
		Integer[] qualities = { 0, -1, 1, 49, 50, 51, Integer.MAX_VALUE, Integer.MIN_VALUE };
		Integer[] sellIns = { -1, 1, 0, 10, 11, 12, 5, 6, 7, Integer.MAX_VALUE, Integer.MIN_VALUE };
		CombinationApprovals.verifyAllCombinations(this::doFoo, names, qualities, sellIns);
	}

	public String doFoo(String name, Integer quality, Integer sellIn) {
		Item item = new Item(name, sellIn, quality);
		Item[] items = { item };
		GildedRose rose = new GildedRose(items);
		rose.updateQuality();
		String string = rose.toString();
		return string;
	}

}
