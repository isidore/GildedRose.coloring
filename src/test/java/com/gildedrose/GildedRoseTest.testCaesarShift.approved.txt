Caesar Shift


0 => ABCDWXYZ abcdwxyz
1 => BCDEXYZA bcdexyza
2 => CDEFYZAB cdefyzab
3 => DEFGZABC defgzabc
4 => EFGHABCD efghabcd
5 => FGHIBCDE fghibcde
6 => GHIJCDEF ghijcdef
7 => HIJKDEFG hijkdefg
8 => IJKLEFGH ijklefgh
9 => JKLMFGHI jklmfghi
10 => KLMNGHIJ klmnghij
11 => LMNOHIJK lmnohijk
12 => MNOPIJKL mnopijkl
13 => NOPQJKLM nopqjklm
14 => OPQRKLMN opqrklmn
15 => PQRSLMNO pqrslmno
16 => QRSTMNOP qrstmnop
17 => RSTUNOPQ rstunopq
18 => STUVOPQR stuvopqr
19 => TUVWPQRS tuvwpqrs
20 => UVWXQRST uvwxqrst
21 => VWXYRSTU vwxyrstu
22 => WXYZSTUV wxyzstuv
23 => XYZATUVW xyzatuvw
24 => YZABUVWX yzabuvwx
25 => ZABCVWXY zabcvwxy
26 => ABCDWXYZ abcdwxyz
27 => BCDEXYZA bcdexyza
28 => CDEFYZAB cdefyzab
29 => DEFGZABC defgzabc
30 => EFGHABCD efghabcd
31 => FGHIBCDE fghibcde
32 => GHIJCDEF ghijcdef
33 => HIJKDEFG hijkdefg
34 => IJKLEFGH ijklefgh
35 => JKLMFGHI jklmfghi
36 => KLMNGHIJ klmnghij
37 => LMNOHIJK lmnohijk
38 => MNOPIJKL mnopijkl
39 => NOPQJKLM nopqjklm
40 => OPQRKLMN opqrklmn
41 => PQRSLMNO pqrslmno
42 => QRSTMNOP qrstmnop
43 => RSTUNOPQ rstunopq
44 => STUVOPQR stuvopqr
45 => TUVWPQRS tuvwpqrs
46 => UVWXQRST uvwxqrst
47 => VWXYRSTU vwxyrstu
48 => WXYZSTUV wxyzstuv
49 => XYZATUVW xyzatuvw
50 => YZABUVWX yzabuvwx
51 => ZABCVWXY zabcvwxy
52 => ABCDWXYZ abcdwxyz
53 => BCDEXYZA bcdexyza
54 => CDEFYZAB cdefyzab
55 => DEFGZABC defgzabc
56 => EFGHABCD efghabcd
57 => FGHIBCDE fghibcde
58 => GHIJCDEF ghijcdef
59 => HIJKDEFG hijkdefg
60 => IJKLEFGH ijklefgh
